File: neetcode-all/array_hashing/majority-element.py
import math 
class Solution:
    def majorityElement(self,nums):
        majorityMap = {}
        for item in nums:
            if item in majorityMap:
                #increase the count
                majorityMap[item] += 1
            #else add
            else: 
                majorityMap[item] = 1

        limit = math.floor(len(nums)/2)
        items_pass = [key for key,val in majorityMap.items() if val>limit]
        return items_pass

print(Solution().majorityElement([2,2,1,1,1,2,2]))

File: neetcode-all/array_hashing/majorityElement.ts
function findMajorityItems(nums: number[]): number | null {
  const majorityMap: { [key: number]: number } = {};

  for (const item of nums) {
    if (item in majorityMap) {
      majorityMap[item] += 1;
    } else {
      majorityMap[item] = 1;
    }
  }

  const limit = Math.floor(nums.length / 2);
  for (const key in majorityMap) {
    if (majorityMap[key] > limit) {
      return parseInt(key, 10);
    }
  }
  return null;
}
